src/app/shell/personal-cabinet/provider/create-provider/create-contacts-form/create-contacts-form.component.ts

```diff
@@ -57,7 +64,38 @@ export class CreateContactsFormComponent implements OnInit, OnDestroy {
     this.passLegalAddressFormGroup.emit(this.LegalAddressFormGroup);
   }
 
+  onSelectedCity(event: any, type: string): void {
+    if (type === 'legal') {
+      this.cityLegal = event.name;
+      this.LegalAddressFormGroup.get('city').reset();
+      this.LegalAddressFormGroup.get('city').setValue(event.name);
+    } else if (type === 'actual') {
+      this.cityActual = event.name;
+      this.ActualAddressFormGroup.get('city').reset();
+      this.ActualAddressFormGroup.get('city').setValue(event.name);
+    }
+  }
+
+  onFocusout(city: City, type: string): void {
+    if (type === 'legal') {
+      if(!this.cityLegalFormControl.value || city.name === Constants.NO_CITY){
+        this.cityLegalFormControl.setValue(null);
+      } else {
+        this.cityLegalFormControl.setValue(this.cityLegal);
+      }
+    } else if (type === 'actual') {
+      if(!this.cityActualFormControl.value || city.name === Constants.NO_CITY){
+        this.cityActualFormControl.setValue(null);
+      } else {
+        this.cityActualFormControl.setValue(this.cityActual);
+      }
```

Comment:
same, let's split this logic
We have two autocomplete in the template so let's they will be handled by their own methods